iceoryx2-ffi/ffi/src/node.rs

```diff
@@ -236,9 +378,95 @@ mod test {
             );
 
             assert_that!(ret_val, eq(IOX2_OK));
+
+            node_handle
+        }
+    }
+
+    #[test]
+    fn basic_node_api_test() {
+        unsafe {
+            let node_handle = create_sut_node();
+
             assert_that!(node_handle, ne(std::ptr::null_mut()));
 
             iox2_node_drop(node_handle);
         }
     }
+
+    #[test]
+    fn basic_node_config_test() {
+        unsafe {
+            let node_handle = create_sut_node();
+            let expected_config = (*node_handle)
+                .node_assume_init::<zero_copy::Service>()
+                .config();
+
+            let config = iox2_node_config(node_handle);
+
+            assert_that!(*config, eq(*expected_config));
+
+            iox2_node_drop(node_handle);
+        }
+    }
+
+    #[derive(Default)]
+    struct NodeListCtx {
+        alive: u64,
+        dead: u64,
+        inaccessible: u64,
+        undefined: u64,
+    }
+
+    extern "C" fn node_list_callback(
+        node_state: iox2_node_state_e,
+        _node_id: iox2_node_id_t,
+        _node_name: iox2_node_name_t,
+        _config: iox2_config_t,
+        ctx: iox2_node_list_callback_ctx,
+    ) -> iox2_callback_progression_e {
+        let ctx = unsafe { &mut *(ctx as *mut NodeListCtx) };
+
+        match node_state {
+            iox2_node_state_e::ALIVE => {
+                ctx.alive += 1;
+            }
+            iox2_node_state_e::DEAD => {
+                ctx.dead += 1;
+            }
+            iox2_node_state_e::INACCESSIBLE => {
+                ctx.inaccessible += 1;
+            }
+            iox2_node_state_e::UNDEFINED => {
+                ctx.undefined += 1;
+            }
+        }
+
+        iox2_callback_progression_e::CONTINUE
+    }
+
+    #[test]
+    fn basic_node_list_test() {
+        unsafe {
+            let mut ctx = NodeListCtx::default();
+            let node_handle = create_sut_node();
+            let config = iox2_node_config(node_handle);
+
+            let ret_val = iox2_nodel_list(
+                iox2_node_type_e::ZERO_COPY,
+                config,
+                node_list_callback,
+                &mut ctx as *mut _ as *mut _,
+            );
+
+            iox2_node_drop(node_handle);
+
+            assert_that!(ret_val, eq(IOX2_OK));
+
+            assert_that!(ctx.alive, eq(1));
+            assert_that!(ctx.dead, eq(0));
```

Comment:
For some reasons, the `Node` is always considered dead ... I also tried with the native Rust API but the result was the same ... I need advice

Reply:
It is a problem with the file lock that - under Linux - can only grabbed once from the process. As soon as the process has the file lock it can lock the file again and again and always gets the lock. It is a mutex like lock only in the inter-process case.

I will try to fix it by acquiring the process id of the node and compare it to my own, if they are the same the node is alive :)